File: TiaCodegen/Commands/Or.cs
﻿using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands
{
    public class Or : BaseOperationOrSignal
    {
        public Or(params IOperationOrSignal[] operationOrSignals)
            : base(operationOrSignals)
        { }

        public override int CreateContactAndFillCardinality(IOperationOrSignal parent)
        {
            int c = 0;
            this.OperationId = parent.OperationId;
            foreach (var op in Children)
            {
                c += op.CreateContactAndFillCardinality(this);
            }

            this.Cardinality = c;
            return this.Cardinality;
        }
    }
}

File: TiaCodegen-ts/src/Commands/Or.ts
import { IOperationOrSignal } from '../Interfaces/IOperationOrSignal';
import { BaseOperationOrSignal } from './BaseOperationOrSignal';

export class Or extends BaseOperationOrSignal {
    constructor(...operationOrSignals: IOperationOrSignal[]) {
        super(...operationOrSignals);
    }

    createContactAndFillCardinality(parent: IOperationOrSignal): number {
        let c = 0;
        this.operationId = parent.operationId;
        for (const op of this.children) {
            c += op.createContactAndFillCardinality(this);
        }
        this.cardinality = c;
        return this.cardinality;
    }
}
